lib/monaco-adapter.js

```diff
@@ -373,8 +373,8 @@ var MonacoAdapter = function () {
         this.trigger('change', op, op);
       }
 
-      /** Reverse iterate all changes */
-      event.changes.reverse().forEach(function (change) {
+      /** Iterate through all changes */
+      event.changes.forEach(function (change) {
```

Comment:
Full disclosure: I maintain this library but was not involved at all in writing it.  I have a feeling there was a good reason to use `reverse()` in the first place.  Do you understand what the reason for that processing was and why it's ok to change it now?

Reply:
I've had similar thoughts while reviewing the code. I do not fully understand the use of `reverse().` If I were to guess, perhaps it was to address the way that Monaco emitted changes in a previous version.

Based on my debugging, reversing the changes would consistently throw an error.

Example (Using the same steps from example 1 from issue #380):
When undoing multiple changes after deleting the last two characters from the word "Hello."

The correct result should be
Hel --*Ctrl + Z*-> Hel + l (offset: 3) + o (offset: 4)

Monaco seems to return the changes in the correct order; reversing the changes would result with:
Hel --*Ctrl + Z*-> Hel + o (offset: 4) + l (offset: 3)

The inconsistency with the offset then throws an error further down the operation chain.
![Screen Shot 2020-09-14 at 9 19 36 AM](https://user-images.githubusercontent.com/35940686/93111412-6be00b00-f66b-11ea-9e92-0f0b6e852f46.jpg)

The same applies to other multi-change events I've tested, such as multi-line commenting (Ctrl + ?), find/replace all occurrences, text-dragging, etc.